sympy/assumptions/ask.py

```diff
@@ -1326,7 +1326,9 @@ def ask(proposition, assumptions=True, context=global_assumptions):
     # direct resolution method, no logic
     res = key(expr)._eval_ask(assumptions)
     if res is not None:
-        return bool(res)
+        if isinstance(res, BooleanAtom):
+            return bool(res)
+        return res
```

Comment:
You are modifying the `ask` function... do you really need this edit?

I don't see any tests on this specific case.

Should `_eval_ask` return some non-boolean valued expression? Are you sure about this edit?